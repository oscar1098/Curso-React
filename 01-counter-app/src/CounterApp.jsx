import {useState} from 'react';
import PropTypes from "prop-types";


export const CounterApp = ( { value } ) => {

    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => {
        setCounter( counter + 1 )
    }

    const handleRest = () => {
        setCounter( counter-1 )
    }

    const handelRestart = () => {
        setCounter( value )
    }

    return ( 
        <>
            <h1>CounterApp</h1>
            <h2>{ counter }</h2>
            <button onClick={  handleAdd }>
                +1
            </button>
            <button onClick={ handleRest } >-1</button>
            <button  aria-label='btn-reset' onClick={ handelRestart } >Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value : PropTypes.number.isRequired
}








