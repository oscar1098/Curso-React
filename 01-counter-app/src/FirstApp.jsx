import PropTypes from 'prop-types';

export const FirstApp = ({title, subTitle, name}) => {

  return (
    <>
    <h1 data-testid="test-title">{title}</h1>
    <p>{subTitle}</p>
    <p>{subTitle}</p>
    <p>{name}</p>
    </>
  );
}

FirstApp.propTypes = {
  name : PropTypes.string,
  subTitle : PropTypes.string,
  title : PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
  name : "Oscar Ardila",
  subTitle : "No hay subtitulo",
  // title : "No hay titulo",
}
