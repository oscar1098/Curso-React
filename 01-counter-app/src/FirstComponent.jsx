const newMessage = 'Oscar Ardila';

export const FirstComponent = () => {

    return (
        <>
            <h1> { newMessage } </h1>
            <h3>Subtitulo</h3>
        </>
    );
}

export const SecondComponent = () => {
    return (
        <p>Lorem ipsum dolor sit amet consectetur, adipisicing elit. Autem sit asperiores nulla fugiat velit consequatur cupiditate, quibusdam possimus porro, facilis natus tenetur totam suscipit recusandae itaque similique aliquam animi repellendus?</p>

    )
}








