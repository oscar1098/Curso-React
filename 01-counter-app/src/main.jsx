import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstComponent, SecondComponent } from './FirstComponent';


ReactDOM.createRoot( document.getElementById('root') ).render(

    <React.StrictMode>
        <FirstComponent/>
        <SecondComponent/>
    </React.StrictMode>
);

