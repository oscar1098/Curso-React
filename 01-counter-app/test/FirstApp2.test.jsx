import { render,screen }  from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Pruebas en <FirstApp2 />', () => { 
    
    const title = 'Hola a todos';
    const subTitle = 'Soy un subtitulo';
    

   test('Debe hacer match con el snapshot', () => { 

       const { container } = render( <FirstApp title = { title } /> );

       expect(container).toMatchSnapshot();
    });

    test('Dede mostrar el mensaje "Hola a todos"', () => { 

        render( <FirstApp title={ title } /> );
        expect( screen.getAllByText ).toBeTruthy();
        // screen.debug();
     });

     test('Debe mostrar el titulo en un h1', () => { 
        
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', {level: 1}).innerHTML ).toContain( title );

     });

     test('Debe de mostrar el subtitulo enviado por prpos', () => { 

        render( 
        <FirstApp 
            title= {title} 
            subTitle={subTitle}
        /> 
        );

        expect( screen.getAllByText( subTitle).length).toBe(2);
      })
 });