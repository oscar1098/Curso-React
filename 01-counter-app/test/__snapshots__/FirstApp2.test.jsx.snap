// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp2 /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola a todos
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Oscar Ardila
  </p>
</div>
`;
